.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,33 @@
+name: Update version on new release branch
+
+on:
+  create:
+
+jobs:
+  update_version:
+    if: github.ref_type == 'branch' && startsWith(github.ref, 'refs/heads/release-v')
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+
+    - name: Extract current version
+      run: |
+        CURRENT_VERSION=$(grep '^version = ' Scarb.toml | sed 's/version = "\(.*\)"/\1/')
+        echo "CURRENT_VERSION=$CURRENT_VERSION" >> $GITHUB_ENV
+
+    - name: Extract new version number
+      run: echo "NEW_VERSION=${GITHUB_REF#refs/heads/release-v}" >> $GITHUB_ENV
+
+    - name: Replace version in files
+      run: |
+        echo "Current version: $CURRENT_VERSION"
+        echo "New version: $NEW_VERSION"
+        find . -type f -not -path '*/\.*' -not -path './CHANGELOG.md' -not -path './docs/package-lock.json' -not -path './RELEASING.md' -exec sed -i "s/$CURRENT_VERSION/$NEW_VERSION/g" {} +
+
+    - name: Auto-commit changes
+      uses: stefanzweifel/git-auto-commit-action@3ea6ae190baf489ba007f7c92608f33ce20ef04a #v4.16.0
+      with:
+        token: ${{ secrets.OPENZEPPELIN_ORG_GITHUB_PAT }}
```

Comment:
The token should be in `actions/checkout`, no? I don't think `token` is available in the auto-commit action itself ([usage](https://github.com/stefanzweifel/git-auto-commit-action?tab=readme-ov-file#usage))

Reply:
why do you say so? this worked in my fork

Reply:
Since I've already glanced at it let me put in my two cents. I didn't try to run the workflow but I can't find the [`token` parameter in the action](https://github.com/stefanzweifel/git-auto-commit-action/blob/3ea6ae190baf489ba007f7c92608f33ce20ef04a/action.yml) so I think this line doesn't influence the action's behaviour.